src/Swarm/Game/CESK.hs

```diff
@@ -288,13 +289,14 @@ idleMachine :: CESK
 idleMachine = Out VUnit emptyStore []
 
 -- | Cancel the currently running computation.
-cancel :: CESK -> CESK
-cancel cesk = Out VUnit s' []
+cancel :: CESK -> IO CESK
+cancel cesk = (\s -> Out VUnit s []) <$> s'
  where
-  s' = resetBlackholes $ getStore cesk
-  getStore (In _ _ s _) = s
-  getStore (Out _ s _) = s
-  getStore (Up _ s _) = s
+  s' = resetBlackholes <$> getStore cesk
+  getStore :: CESK -> IO Store
+  getStore (In _ _ s _) = pure s
+  getStore (Out v s _) = (case v of VAsync (GAsync _ as) -> Async.cancel as; _ -> pure ()) >> pure s
```

Comment:
This isn't guaranteed to cancel all currently running async computations, because there are lots of other places `VAsync` values could be hiding: in the environment, in a closure, ...  I don't know how important this is.  Perhaps not that important, since if an async computation keeps running but nothing ever asks for its value it only wastes some CPU cycles, it won't cause any crashes or anything like that.  If it is important, we would have to do something like put references to currently outstanding async computations in the store, so that we can find them all when we want to cancel them.

Reply:
I think I do not want to cancel *all async computations* - the ones currently being run by base are enough.

We could store the async values in GameState and add some way to cancel all of them in the UI.